build-system/babel-plugins/babel-plugin-transform-inline-isenumvalue/index.js

```diff
@@ -0,0 +1,108 @@
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @param {*} babel
+ */
+module.exports = function (babel) {
+  const {types: t, template} = babel;
+
+  function maybeTransformAssertEnumValue(path) {
+    const callee = path.get('callee');
+    const object = callee.get('object');
+    const property = callee.get('property');
+    if (
+      !callee.isMemberExpression() ||
+      !object.isCallExpression() ||
+      !property.isIdentifier({name: 'assertEnumValue'})
+    ) {
+      return false;
+    }
+    const objectCallee = object.get('callee');
+    const assertFnName = `${objectCallee.node.name}Assert`;
+    const [enumId, subject, enumNameNode] = path.node.arguments;
+    if (assertFnName === 'devAssert') {
+      path.replaceWith(t.cloneNode(subject));
+      return true;
+    }
+    const {specifiers} = path.scope.getBinding(objectCallee.node.name).path
+      .parent;
+    let specifier = specifiers.find(
+      ({imported}) =>
+        imported.type === 'Identifier' && imported.name === assertFnName
+    );
+    if (!specifier) {
+      specifier = t.importSpecifier(
+        t.identifier(assertFnName),
+        t.identifier(assertFnName)
+      );
+      specifiers.push(specifier);
+    }
+    const enumName = enumNameNode?.value || enumNameNode?.name || 'enum';
+    path.replaceWith(
+      template.expression`
+        (
+          ${t.identifier(specifier.local.name)}(
+            isEnumValue(${t.cloneNode(enumId)}, ${t.cloneNode(subject)}),
+            ${t.stringLiteral(`Unknown ${enumName} value: "`)} +
+              ${t.cloneNode(subject)} + '"'
+          ),
+          ${t.cloneNode(subject)}
+        )
+      `()
+    );
+    return true;
+  }
+
+  return {
+    name: 'transform-inline-isenumvalue',
+    visitor: {
+      CallExpression(path) {
+        // Replace x().assertEnumValue() calls with xAssert(isEnumValue(), ...).
+        // The replacement is handled on a subsequent pass.
+        if (maybeTransformAssertEnumValue(path)) {
+          return;
+        }
+
+        // isEnumValue()
+        const callee = path.get('callee');
+        if (!callee.isIdentifier({name: 'isEnumValue'})) {
+          return;
+        }
+        const enumArg = path.get('arguments.0');
+        const {confident, value} = enumArg.evaluate();
+        if (!confident) {
+          // throw path.buildCodeFrameError('Cannot evaluate. Is it imported?');
```

Comment:
Below is [a comment](https://github.com/ampproject/amphtml/pull/34465/files#r636239397) about a possible linter. If we don't use a new lint rule, we can get strict and error out here instead.

Reply:
@jridgewell What do you think? Lint it or error out?

Reply:
I would use a Linter

Reply:
Or, if you want to get real advanced, you could spin up a Babel parser on the imported file, and have it evaluate on the imported file's AST instead. Maybe too much work…

Reply:
I feel better about the added complexity since we removed `assertEnumValue`, so I'm resolving the import now.